masonry/src/passes/update.rs

```diff
@@ -498,10 +499,42 @@ fn update_new_widgets(
         let mut ctx = RegisterCtx {
             widget_state_children: state.children.reborrow_mut(),
             widget_children: widget.children.reborrow_mut(),
+            #[cfg(debug_assertions)]
+            registered_ids: Vec::new(),
         };
         // The widget will call `RegisterCtx::register_child` on all its children,
         // which will add the new widgets to the arena.
         widget.item.register_children(&mut ctx);
+
+        #[cfg(debug_assertions)]
+        {
+            let children_ids = widget.item.children_ids();
+            for child_id in ctx.registered_ids {
+                if !children_ids.contains(&child_id) {
```

Comment:
I think that if we used `find` and `swap_remove` here, we would get the second check for free (just any items remaining in `children_ids`).

Reply:
I'd rather have code be maximally clear.

What I'd like to do is add an automatic escape hatch for cases where `children_ids` is large, so we can skip checking those (under the assumption that if there's a bug in child handling, it probably shows up with a small number of children too).